test/unit/math/mathOps/include/Functor.hpp

```diff
@@ -0,0 +1,288 @@
+//! @param ARITY Enum-type can be one ... n
+//! @param STD_OP Function used for the host side, e.g. std::abs
+//! @param ALPAKA_OP Function used for the device side, e.g. alpaka::math::abs.
+//! @param ... List of Ranges. Needs to match the arity.
+#define ALPAKA_TEST_MATH_OP_FUNCTOR( NAME, ARITY, STD_OP, ALPAKA_OP, ... )      \
+  struct NAME                                                                   \
+  {                                                                             \
+    /* ranges is not a constexpr, so that it's accessible via for loop*/        \
+    static constexpr Arity arity = ARITY;                                       \
+    static constexpr size_t arity_nr = static_cast<size_t>(ARITY);              \
+    const Range ranges[ arity_nr ] = {__VA_ARGS__};                             \
+                                                                                \
+    ALPAKA_NO_HOST_ACC_WARNING                                                  \
+    template<typename TAcc,                                                     \
+             typename... TArgs,                                                 \
+             /* SFINAE: Enables if called from device. */                       \
+             typename std::enable_if<                                           \
+                 !std::is_same<TAcc, std::nullptr_t>::value,                    \
+                 int>::type = 0>                                                \
+    ALPAKA_FN_ACC                                                               \
+    auto execute(                                                               \
+        TAcc const & acc,                                                       \
+        TArgs const & ... args ) const                                          \
+    -> decltype( ALPAKA_OP(acc, args... ) )                                     \
+    {                                                                           \
+        return ALPAKA_OP(acc, args... );                                        \
+    }                                                                           \
+                                                                                \
+    ALPAKA_NO_HOST_ACC_WARNING                                                  \
+    template<                                                                   \
+        typename TAcc = std::nullptr_t,                                         \
+        typename... TArgs,                                                      \
+             /* SFINAE: Enables if called from host. */                         \
+             typename std::enable_if<                                           \
+                 std::is_same< TAcc, std::nullptr_t>::value,                    \
+                 int>::type = 0>                                                \
+    ALPAKA_FN_HOST                                                              \
```

Comment:
clang as CUDA compiler complains: `error: reference to __host__ function 'execute<nullptr_t, double, 0>' in __host__ __device__ function`